### PR TITLE
Np 47307 CsvTransformer, create files for types identifiers, contributors, affiliations and fundings

### PR DESCRIPTION
**Pull request by monaullah**
Jira task: NP-46747 Database-dumps for initial import

Part two:

- For each key batch, `CsvBulkTransformer` creates five csv files (publication, affiliation, contributor, identifier, funding) and persists in separate folders in s3 bucket `data-report-csv-export`
- Removed compression from `CsvBulkTransformer` since consumer preferred without

Next steps:
- Support nvi export

### CODE REVIEW

File: bulk-export/src/test/java/no/sikt/nva/data/report/api/export/CsvTransformerTest.java [hunk 1]
```diff
@@ -69,43 +74,51 @@ void setUp() {
...
-                                                      CONTRIBUTOR_IDENTIFIER);
-        var expectedContent = testData.getPublicationResponseData();
+        var actualContent = attempt(() -> sortResponse(CSV, getActualPersistedFile(reportType), PUBLICATION_ID,
+                                                       CONTRIBUTOR_IDENTIFIER)).orElseThrow();
+        var expectedContent = getExpectedResponseData(reportType, testData);
         assertEquals(expectedContent, actualContent);
     }
 
+    @Test
+    void shouldWriteCsvFilesForAlleReportTypesToSpecificFolderInExportBucket() throws IOException {
```
Comment (truhacevkir) [inline bulk-export/src/test/java/no/sikt/nva/data/report/api/export/CsvTransformerTest.java:92]: *All

File: report-api/src/main/java/no/sikt/nva/data/report/api/fetch/model/ReportRequest.java [hunk 2]
```diff
@@ -124,4 +112,16 @@ private static String extractReportType(RequestInfo requestInfo) {
...
+    private void validate() throws BadRequestException {
+        if (this.offset < 0) {
+            throw new BadRequestException("Offset cannot be less than zero");
+        }
+        if (this.pageSize < 0) {
+            throw new BadRequestException("Page size cannot be less than zero");
+        }
+        if (this.after.isAfter(this.before)) {
+            throw new BadRequestException("Logically, 'before should be after the value of 'after'");
+        }
```
Comment (truhacevkir) [inline report-api/src/main/java/no/sikt/nva/data/report/api/fetch/model/ReportRequest.java:125]: Is it correct? Shouldn't it be like this?

```
if (this.after.isBefore(this.before)) {
    throw new BadRequestException("'after' should not be before 'before'");
}
```

Comment (monaullah) [reply report-api/src/main/java/no/sikt/nva/data/report/api/fetch/model/ReportRequest.java:125]: https://github.com/BIBSYSDEV/nva-data-report-api/pull/2#discussion_r1407320364 😄 Can change this, but in separate PR

**Review by truhacevkir — APPROVED** [review]

File: report-api/src/main/java/no/sikt/nva/data/report/api/fetch/model/ReportRequest.java (hunk 2 above)
Comment (truhacevkir) [reply report-api/src/main/java/no/sikt/nva/data/report/api/fetch/model/ReportRequest.java:125]: I would also change error message to not repeat "..be before 'before'".
`Usually, in time comparisons 'after' is a moment later in time than 'before'.`

**Review by brinxmat — APPROVED** [review]

**Pull request merged**